engine/core/entity/Entity.cpp

```diff
@@ -14,9 +14,9 @@
 #include "EntitySystem.h"
 
 // Static member initialisation
-const String Entity::ENTITY_NAME("Entity");
+const Siege::String Entity::ENTITY_NAME("Entity");
```

Comment:
Let's avoid adding this name spacing for any usage of `String` across the engine by wrapping each class within its appropriate namespace.

Reply:
This is only the case right now since there's a future PR that does this. Doing this now would cannibalise that PR.